hq/app/logic/SnykDisplay.scala

```diff
@@ -1,81 +1,63 @@
-    (Json.parse(s) \ "projects").asOpt[List[SnykProject]]
-  } catch {
-    case _: JsonParseException => None
-  }
+  def parseJsonToOrganisationList(s: String): Attempt[List[SnykOrganisation]] = parseJsonToObject("organisations", s, body => {(Json.parse(body) \ "orgs").validate[List[SnykOrganisation]]} )
 
-  def parseProjectVulnerabilities(projects: List[String])(implicit ec:ExecutionContext): Attempt[List[SnykProjectIssues]] = {
-    val projectVulnerabilitiesList = projects.map( s => {
-      val projectVulnerabilities = parseJsonToProjectVulnerabilities(s)
-      val error = parseJsonToError(s)
-      val f = Failure(s"Unable to find project vulnerabilities from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
-      Attempt.fromOption(projectVulnerabilities, FailedAttempt(f))
-    })
-    Attempt.traverse(projectVulnerabilitiesList)(a => a)
-  }
+  def parseJsonToProjectIdList(s: String): Attempt[List[SnykProject]]         = parseJsonToObject("project ids", s, body => (Json.parse(body) \ "projects").validate[List[SnykProject]] )
 
-  private def parseJsonToProjectVulnerabilities(s: String) = try {
-    Json.parse(s).asOpt[SnykProjectIssues]
-  } catch {
-    case _: JsonParseException => None
-  }
+  def parseProjectVulnerabilitiesBody(s: String): Attempt[SnykProjectIssues]  = parseJsonToObject("project vulnerabilities", s, body => Json.parse(body).validate[SnykProjectIssues])
 
-  def linkToSnykProject(snykProjectIssues: SnykProjectIssues, queryString: Option[String]): String = {
-    snykProjectIssues.project match {
-      case Some(project: SnykProject) if project.organisation.nonEmpty =>
-        s"https://snyk.io/org/${project.organisation.get.name}/project/${project.id}/${queryString.getOrElse("")}"
-      case _ => ""
-    }
-  }
+  def parseProjectVulnerabilities(bodies: List[String])(implicit ec: ExecutionContext): Attempt[List[SnykProjectIssues]] = Attempt.traverse(bodies)(parseProjectVulnerabilitiesBody)
 
-  def labelOrganisations(projects: List[SnykProject], snykOrg: SnykOrganisation): List[SnykProject] = {
-    projects.map(project => project.copy(organisation = Some(snykOrg)))
+  def linkToSnykProject(snykProjectIssues: SnykProjectIssues, queryString: Option[String]): String = snykProjectIssues.project match {
+    case Some(project) if project.organisation.nonEmpty =>
+      s"https://snyk.io/org/${project.organisation.get.name}/project/${project.id}/${queryString.getOrElse("")}"
+    case _ => ""
   }
 
-  def labelProjects(projects: List[SnykProject], responses: List[SnykProjectIssues]): List[SnykProjectIssues] = {
-    projects.zip(responses).map(a => a._2.copy(project = Some(a._1)))
-  }
+  def labelOrganisations(orgAndProjects: List[((SnykOrganisation, String), List[SnykProject])]): List[SnykProject] =
+    orgAndProjects.flatMap{ case ((organisation, _), projects) => projects.map(project => project.copy(organisation = Some(organisation)))}
```

Comment:
Long line! Space after the `flatMap` too. It's either a simple inline fn tight in with `()`, or a more complex one spaced out with `{}`

```scala
orgAndProjects.flatMap { case ((organisation, _), projects) =>
  projects.map(project => project.copy(organisation = Some(organisation)))
}
```
`project` could be underscore here now if you like, since we're right next to `projects.map` so we know it's a project.
```scala
orgAndProjects.flatMap { case ((organisation, _), projects) =>
  projects.map(_.copy(organisation = Some(organisation)))
}
```